src/widgets/stepUp/mfaStepUpWidget.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {createMultiViewWidget} from "../../components/widget/widget";
+import {deepDefaults} from "../../helpers/deepDefaults";
+import {createForm} from "../../components/form/formComponent";
+import radioboxField from "../../components/form/fields/radioboxField";
+import {Info, Intro} from "../../components/miscComponent";
+import {simpleField} from "../../components/form/fields/simpleField";
+import {toQueryString} from "../../helpers/queryString";
+
+const StartStepUpMfaButton = createForm({
+    prefix: 'r5-mfa-start-step-up-',
+    submitLabel: 'mfa.stepUp.start',
+});
+
+const VerificationCodeInputForm = createForm({
+    prefix: 'r5-passwordless-sms-',
+    fields: [
+        simpleField({
+            key: 'verification_code',
+            label: 'verificationCode',
+            type: 'text'
+        })
+    ]
+});
+
+const StartPasswordlessForm = createForm({
+    prefix: 'r5-mfa-start-passwordless',
+    fields({ options }) {
+        return [
+            radioboxField({
+                key: 'authType',
+                options
+            }),
+        ]
+    }
+})
+class MainView extends React.Component {
+    onGetStepUpToken = _ => {
+        return this.props.apiClient.getMfaStepUpToken({
+            options: this.props.auth,
+            accessToken: this.props.accessToken
+        })
+    }
+    render() {
+        return <div>
+            <StartStepUpMfaButton
+                handler={this.onGetStepUpToken}
+                onSuccess={data => this.props.goTo('fa-selection', {...data})}
+                />
+        </div>
+    }
+}
+
+class FaSelectionView extends React.Component {
+    onChooseFa = factor => {
+        return this.props.apiClient.startPasswordless({
+            ...factor,
+            stepUp: this.props.token,
+        }).then(resp =>
+            ({...resp, ...factor})
+        )
+    }
+
+    render() {
+        const { amr, showIntro, i18n } = this.props
+        return <div>
+            {showIntro && <Intro>{i18n('mfa.select.factor')}</Intro>}
+            <StartPasswordlessForm
+                options={amr.map(factor => ({ key: factor, value: factor, label: factor}))}
```

Comment:
Is that preselected when there only is one factor available?

Reply:
You are right I changed the behaviour to preselect the only factor available when there is only one second factor.